safety/auth/utils.py

```diff
@@ -425,3 +426,20 @@ def send(self, request: requests.PreparedRequest, **kwargs: Any) -> requests.Res
         """
         request.headers.pop("Authorization", None)
         return super().send(request, **kwargs)
+    
+    
+@lru_cache(maxsize=1)
+def is_jupyter_notebook() -> bool:
+    """
+    Check if the code is being executed inside a Jupyter Notebook.
+    
+    Returns:
+        bool: True if running in a Jupyter notebook, False otherwise.
+    """
+    try:
+         from IPython import get_ipython
```

Comment:
Does it make sense to move this import to the top of the file?

Reply:
Yes

Reply:
IPython will only be present in jyupter notebook env.